libs/application/templates/parental-leave/src/fields/PeriodPercentage/PeriodPercentage.tsx

```diff
@@ -94,13 +98,45 @@ export const PeriodPercentage: FC<PeriodPercentageField> = ({
         label: `${maxPercentage - index}%`,
       }))
 
+    const periodLengthWithMaxPercentage = calculatePeriodLength(
+      start,
+      end,
+      maxPercentage / 100,
+    )
+
+    if (
+      periodLengthWithMaxPercentage < remainingRights &&
+      maxPercentage < 100
+    ) {
+      setCanChooseRemainingDays(true)
+      const max = `${maxPercentage + 1}`
+      setMaxPercentageValue(max)
+
+      options.splice(0, 0, {
+        value: max,
+        label: formatMessage(
+          parentalLeaveFormMessages.duration.fullyUsedRatio,
+          { maxPercentage: max },
+        ),
+      })
+    }
```

Comment:
Minor: it's a bit weird to use `useEffect` to initialize component state from available data (this whole function?). Since it performs a redundant render, possibly with a flash of wrong data.

In the future, you might want to have this kind of logic in the component or a utility function. If you want to make sure to only run this once (instead of during each render), `useMemo` can help.